quick_select/quick_select.c

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+
+/*
+ * partition function
+ * ar        : array on which partitioning has to be done
+ * left      : left index of the partitioning subarray
+ * right     : right index of the partitioning subarray
+ * pivot_idx : pivot index from which partition has to done
+ * store_idx : index of the last element of the left subarray
+ * return    : int  
+ */
+int partition(int *ar, int left, int right, int pivot_idx) {
+   int pivot_value = ar[pivot_idx];
+   int temp = ar[right];
+   ar[right] = ar[pivot_idx];
+   ar[pivot_idx] = temp;
+
+   int store_idx = left;
+   while (left < right) {
+      if (ar[left] < pivot_value) {
+         temp = ar[store_idx];
+         ar[store_idx] = ar[left];
+         ar[left] = temp;
+         store_idx++;
+      }
+      left++;
+   }
+   temp = ar[right];
+   ar[right] = ar[store_idx];
+   ar[store_idx] = temp;
+   return store_idx;
+}
+
+/*
+ * Quick Select function
+ * left        : left index of the subarray
+ * right       : right index of the subarray
+ * pos         : position to find the element using quick sort
+ * pivot_index : pivot index 
+ * return      : int
```

Comment:
But it looks strange. We have a list of parameters and out of nowhere we see the return type. What I see is that the return type is already indicated very clearly in `C` (`int quick_select(...) {}`). It would be better to add **what** it returns instead of return type.